### PR TITLE
feat(aws): Add MFA flag if it is required by AWS IAM Entity

### PR DESCRIPTION
**Pull request by senyberg**
### Context

Currently you cannot assume roles in AWS if the roles are using MFA

### Description

Add "--mfa" flag which asks user for MFA ARN + Code to authenticate

### Acceptance Criteria

1. Run Prowler with user that requires MFA without --mfa -> Access Denied
2. Run Prowler with user that requires MFA with --mfa-> No errors
3. Run Prowler assuming a role that requires MFA without --mfa -> Access Denied
4. Run Prowler assuming a role that requires MFA with --mfa -> No errors

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/aws_provider.py [hunk 1]
```diff
@@ -91,11 +91,22 @@ def assume_role(session: session.Session, assumed_role_info: AWS_Assume_Role) ->
...
         # else assume the role without the external id
         else:
-            assumed_credentials = sts_client.assume_role(
-                RoleArn=assumed_role_info.role_arn,
-                RoleSessionName="ProwlerProAsessmentSession",
-                DurationSeconds=assumed_role_info.session_duration,
-            )
+            if assumed_role_info.mfa_enabled:
+                mfa_ARN = input("Enter ARN of MFA: ")
+                mfa_TOTP = input("Enter MFA code: ")
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py]: Does it makes sense to use `string.strip()` to remove leading and trailing spaces?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Hi @senyberg, this is an awesome PR!!

Some comments/questions about this in order to merge it:
- Could you update the Prowler documentation [here](https://docs.prowler.cloud/en/latest/tutorials/aws/role-assumption/) to include this new flag?
- Could you include a new test for the `parser.py` like these https://github.com/prowler-cloud/prowler/blob/master/tests/lib/cli/parser_test.py#L673?

Thanks again for using Prowler 🚀

**Comment by senyberg** [comment]
I added the test and strip() to the input. I have a couple of questions:

- Where do you update the documentation?
- Do you want the flag to be --mfa, or would you like it to be more specific to the role assumption? For example --role-mfa?

File: tests/lib/cli/parser_test.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import pytest
 
 from prowler.lib.cli.parser import ProwlerArgumentParser
+from prowler.lib.logger import logger
```
Comment (jfagoagas) [inline tests/lib/cli/parser_test.py]: The linter is raising that this value is unused, please remove it if so.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

**Comment by jfagoagas** [comment]
> I added the test and strip() to the input. I have a couple of questions:

Awesome!

> * Where do you update the documentation?

The tutorial for the Role assumption in Boto3 is here https://github.com/prowler-cloud/prowler/blob/master/docs/tutorials/aws/role-assumption.md

> * Do you want the flag to be --mfa, or would you like it to be more specific to the role assumption? For example --role-mfa?

Good question, we could use `--role-mfa` to be more specific. Please change the references in the code and tests.

Thank you for your contribution!

File: tests/lib/cli/parser_test.py (hunk 2 above)
Comment (senyberg) [reply tests/lib/cli/parser_test.py]: Yes, forgot it there. I will remove it. Just need to fix a bunch of other tests

Comment (jfagoagas) [reply tests/lib/cli/parser_test.py]: If you need help with the tests tell me. Thanks!

**Review by MrCloudSec — APPROVED** [review]

**Review by jfagoagas — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀 🚀

**Pull request merged**